src/test_suite/peekable_sequence.ts

```diff
@@ -0,0 +1,47 @@
+export class PeekableSequence<T> {
+    iterator: IterableIterator<T>;
+    current: IteratorResult<T>;
+
+    constructor(iterator: IterableIterator<T>) {
+        this.iterator = iterator;
+        this.current = this.iterator.next();
+    }
+
+    peek(): T {
+        if (!this.current.done) {
+            return this.current.value;
+        } else {
+            throw TypeError('PeekableSequence<T>.peek(): at end of stream.');
+        }
+    }
+
+    get(): T {
+        if (!this.current.done) {
+            const value = this.current.value;
+            this.current = this.iterator.next();
+            return value;
+        } else {
+            throw TypeError('PeekableSequence<T>.get(): at end of stream.');
+        }
+    }
+
+    atEOS(): boolean {
```

Comment:
Is EOS end of stream?

Reply:
EOS does stand for End of Stream. In general, we avoid abbreviations unless they are in common use. EOF is pretty common. Perhaps EOS is too obscure?

Reply:
I added a comment to the code.

Reply:
I realized it was end of stream from reading the rest of the code, maybe it is common to most - it's up to you if you want to rename it but it's not a big deal